spec/factories/courses.rb

```diff
@@ -208,14 +208,16 @@
       to_create { |instance| instance.save(validate: false) }
     end
 
-    trait :published_with_primary_maths do
+    trait :unpublished_with_primary_maths do
       transient do
-        identifier { "published_with_primary_mathematics" }
+        identifier { "unpublished_with_primary_mathematics" }
       end
 
       name { "#{identifier} course" }
 
       provider { find_or_create :provider, :published_scitt }
+      site_statuses { [build(:site_status, :new)] }
```

Comment:
This will need to be associated with the correct provider. There's a few ways to do this:

1. add functionality to the `site_status` or `site` factory so that if the provider wasn't specified, to introspect the course and use it's provider, perhaps as a separate trait like `:auto_provider`
2. add the course's provider to the attributes here
3. add a trait to the `provider` factory to be able to add a site to it

... ya, this will warrant a little bit more thinking.

Reply:
Change made here - https://github.com/DFE-Digital/manage-courses-backend/pull/1097/commits/65c368d95828c9c3b04fe6d3ffbf13d46700ec03